### PR TITLE
#10 Debug Workflows

### LINKED ISSUE #10: Debug Workflows
This issue addresses the content of the Debug Workflows page. This page should present the tools available in shift to debug workflows to the user. This includes:
- Explaining visual clues when a node errors
- Show traceback
- Add breakpoints
- Inspect logger

Screenshots should be added where needed

### PR DESCRIPTION
**Pull request by juliavisserinbibo**
## Issue
Closes #10

## Description
This PR addresses the content of the Debug Workflows page. This page should present the tools available in shift to debug workflows to the user. This includes:
- Explaining visual clues when a node errors
- Show traceback
- Inspect logger
- Add breakpoints
- Stop On Error option

This PR comes with the addition of new gifs to visualize each subsection.

Notes:
- The Logger widget will be properly linked when the UI Overview page is finished.

### CODE REVIEW

File: docs/getting_started/basics/debug.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Debug Workflows
+
+While working on a workflow, debugging may be necessary to ensure the graph execution produces the expected results. This page outlines Shift's features designed for workflow debugging.
+
+
+## Erred Nodes & Traceback
```
Comment (hectorjimenezinbibo) [inline docs/getting_started/basics/debug.md:6]:
```suggestion
## Errored Nodes & Traceback
```

**Review by hectorjimenezinbibo — CHANGES_REQUESTED** [review]

File: docs/images/traceback.gif
Comment (JorgeHernandezibanezinbibo) [inline docs/images/traceback.gif:1]: Should we wait to adress https://github.com/Inbibo/shift/issues/1083 to record this Gif again?

**Review by JorgeHernandezibanezinbibo — COMMENTED** [review]
One question about the gif

File: docs/getting_started/basics/debug.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# Debug Workflows
+
+While working on a workflow, debugging may be necessary to ensure the graph execution produces the expected results. This page outlines Shift's features designed for workflow debugging.
+
+
+## Errored Nodes & Traceback
+When a node does not execute successfully it will turn red and the workflow execution will be interrupted. In such case, the node will store the traceback generated by the error. To inspect the traceback, `Right Click` on the node and select the *Traceback* option. This will pop up a dialog with the error traceback content. 
+
+![Breakpoint](../../images/traceback.gif)
```
Comment (albertochimentiinbibo) [inline docs/getting_started/basics/debug.md:9]:
```suggestion
![Traceback](../../images/traceback.gif)

```

File: docs/getting_started/basics/debug.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+While working on a workflow, debugging may be necessary to ensure the graph execution produces the expected results. This page outlines Shift's features designed for workflow debugging.
+
+
+## Errored Nodes & Traceback
+When a node does not execute successfully it will turn red and the workflow execution will be interrupted. In such case, the node will store the traceback generated by the error. To inspect the traceback, `Right Click` on the node and select the *Traceback* option. This will pop up a dialog with the error traceback content. 
+
+![Breakpoint](../../images/traceback.gif)
+
+Once the root of the error is fixed, the node will turn blue and its traceback will be cleared when it gets recomputed.
```
Comment (albertochimentiinbibo) [inline docs/getting_started/basics/debug.md:11]:
```suggestion
Once the error is fixed, for example by changing a connection or a plug value, the node will turn blue and its traceback will be cleared when it gets recomputed. 

```

File: docs/getting_started/basics/debug.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+## Errored Nodes & Traceback
+When a node does not execute successfully it will turn red and the workflow execution will be interrupted. In such case, the node will store the traceback generated by the error. To inspect the traceback, `Right Click` on the node and select the *Traceback* option. This will pop up a dialog with the error traceback content. 
+
+![Breakpoint](../../images/traceback.gif)
+
+Once the root of the error is fixed, the node will turn blue and its traceback will be cleared when it gets recomputed. 
+
+### Logger
+The error traceback information can be consulted in the [Logger widget](../ui_overview/#logger) too. The logger also reports warnings raised by nodes, and other Shift messages such as plug connection errors or other general logs related to the workflow or the application itself. 
+To learn more about the Logger's messages and configuration, please refer to this [page](../../reference/logger.md).
```
Comment (albertochimentiinbibo) [inline docs/getting_started/basics/debug.md:15]:
```suggestion
The error traceback information can be consulted in the [Logger widget](../ui_overview/#logger) too. The logger also reports warnings raised by nodes, and other Shift messages such as plug connection errors or general logs related to the workflow or the application itself. 
To learn more about the Logger's messages and configuration, please refer to this [page](../../reference/logger.md). 

```

File: docs/getting_started/basics/debug.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+Once the root of the error is fixed, the node will turn blue and its traceback will be cleared when it gets recomputed. 
+
+### Logger
+The error traceback information can be consulted in the [Logger widget](../ui_overview/#logger) too. The logger also reports warnings raised by nodes, and other Shift messages such as plug connection errors or other general logs related to the workflow or the application itself. 
+To learn more about the Logger's messages and configuration, please refer to this [page](../../reference/logger.md). 
+
+## Control Workflow Execution
+
+Shift provides many ways of executing workflows in a modular way. For instance, the graph can be executed step by step with the [*Execute Next*](../execute/#execute-next) feature, or execute specific nodes via the [*Execute Selected*](../execute/#execute-selected) feature. These options allow a more precise control over the workflow's execution and enables to inspect the nodes' results during the process.
```
Comment (albertochimentiinbibo) [inline docs/getting_started/basics/debug.md:19]:
```suggestion
Shift provides many ways of executing workflows in a modular way. For instance, the graph can be executed step by step with the [*Execute Next*](../execute/#execute-next) feature, or execute specific nodes via the [*Execute Selected*](../execute/#execute-selected) feature. These options grant a more precise control over the workflow's execution and enable gradual inspection of the nodes' results during the full graph execution.

```

File: docs/getting_started/basics/debug.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+## Errored Nodes & Traceback
+When a node does not execute successfully it will turn red and the workflow execution will be interrupted. In such case, the node will store the traceback generated by the error. To inspect the traceback, `Right Click` on the node and select the *Traceback* option. This will pop up a dialog with the error traceback content. 
+
+![Breakpoint](../../images/traceback.gif)
+
+Once the root of the error is fixed, the node will turn blue and its traceback will be cleared when it gets recomputed. 
+
+### Logger
+The error traceback information can be consulted in the [Logger widget](../ui_overview/#logger) too. The logger also reports warnings raised by nodes, and other Shift messages such as plug connection errors or other general logs related to the workflow or the application itself.
```
Comment (albertochimentiinbibo) [inline docs/getting_started/basics/debug.md:14]:
```suggestion
### Logger

The error traceback information can be consulted in the [Logger widget](../ui_overview/#logger) too. The logger also reports warnings raised by nodes, and other Shift messages such as plug connection errors or other general logs related to the workflow or the application itself. 

```

File: docs/getting_started/basics/debug.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+### Logger
+The error traceback information can be consulted in the [Logger widget](../ui_overview/#logger) too. The logger also reports warnings raised by nodes, and other Shift messages such as plug connection errors or other general logs related to the workflow or the application itself. 
+To learn more about the Logger's messages and configuration, please refer to this [page](../../reference/logger.md). 
+
+## Control Workflow Execution
+
+Shift provides many ways of executing workflows in a modular way. For instance, the graph can be executed step by step with the [*Execute Next*](../execute/#execute-next) feature, or execute specific nodes via the [*Execute Selected*](../execute/#execute-selected) feature. These options allow a more precise control over the workflow's execution and enables to inspect the nodes' results during the process.
+
+### Breakpoints
+Additionally, Shift allows to add breakpoints to the nodes. When a node is flagged with a breakpoint, the execution of the graph will stop once it reaches that node. To add a breakpoint, `Right Click` on the node and select the *Breakpoint* option. To remove it, follow the same steps.
```
Comment (albertochimentiinbibo) [inline docs/getting_started/basics/debug.md:22]:
```suggestion
### Breakpoints

Additionally, Shift allows to add breakpoints to the nodes. When a node is flagged with a breakpoint, the execution of the graph will stop when it reaches that node. To add a breakpoint, `Right Click` on the node and select the *Breakpoint* option. To remove it, click again on the same context menu option. 

```

File: docs/getting_started/basics/debug.md [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
+### Breakpoints
+Additionally, Shift allows to add breakpoints to the nodes. When a node is flagged with a breakpoint, the execution of the graph will stop once it reaches that node. To add a breakpoint, `Right Click` on the node and select the *Breakpoint* option. To remove it, follow the same steps. 
+
+![Breakpoint](../../images/breakpoint.gif)
+
+
+### Stop Execution on Error
+By default, if a node errors the execution of the workflow will be interrupted. While this prevents the workflow from propagating unwanted data, this behavior can also be disabled if desired. 
+To allow the execution of a workflow to continue even if there are errors, go to the top menu a do `Execute -> Stop on Error`. To revert this change, follow the same steps.
```
Comment (albertochimentiinbibo) [inline docs/getting_started/basics/debug.md:29]: Good practice with Markdown files is to leave an empty line at the end and an empty line between title-corpus or code-corpus. Also no need to specify how to revert the change in my opinion.

```suggestion
### Stop Execution on Error

By default, if a node errors the execution of the workflow will be interrupted. While this prevents the workflow from propagating unwanted data, this behavior can also be disabled if desired. 
To allow the execution of a workflow to continue even if there are errors, go to the top menu a do `Execute -> Stop on Error`.

```

**Review by albertochimentiinbibo — CHANGES_REQUESTED** [review]

File: docs/getting_started/basics/debug.md (hunk 6 above)
Comment (leegeertseninbibo) [reply docs/getting_started/basics/debug.md:15]: @albertochimentiinbibo Markdown does not care about empty lines after a title

**Review by hectorjimenezinbibo — APPROVED** [review]

File: docs/images/traceback.gif
Comment (juliavisserinbibo) [reply docs/images/traceback.gif:1]: Yes

File: docs/getting_started/basics/debug.md (hunk 8 above)
Comment (juliavisserinbibo) [reply docs/getting_started/basics/debug.md]: Done

File: docs/getting_started/basics/debug.md (hunk 4 above)
Comment (juliavisserinbibo) [reply docs/getting_started/basics/debug.md]: Done

File: docs/getting_started/basics/debug.md (hunk 5 above)
Comment (juliavisserinbibo) [reply docs/getting_started/basics/debug.md]: Done

File: docs/getting_started/basics/debug.md (hunk 6 above)
Comment (juliavisserinbibo) [reply docs/getting_started/basics/debug.md]: Done

File: docs/getting_started/basics/debug.md (hunk 7 above)
Comment (juliavisserinbibo) [reply docs/getting_started/basics/debug.md]: Done

File: docs/images/traceback.gif
Comment (juliavisserinbibo) [reply docs/images/traceback.gif:1]: Done

**Review by albertochimentiinbibo — APPROVED** [review]

File: docs/images/traceback.gif
Comment (JorgeHernandezibanezinbibo) [reply docs/images/traceback.gif:1]: @juliavisserinbibo I still see the dialog without parent, Is a git problem or the change is not pushed? 
image

Comment (juliavisserinbibo) [reply docs/images/traceback.gif:1]: I did not push haha sorry
Now it's pushed
